File: uva10931/uva10931.c
/*#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int count_parity(int I) {
    int binary[32];
    int count = 0;
    int index = 0;

    while (I > 0) {
        binary[index++] = I % 2;
        count += I % 2;
        I /= 2;
    }

    printf("The parity of ");
    for (int i = index - 1; i >= 0; i--) {
        printf("%d", binary[i]);
    }
    printf(" is %d (mod 2).\n", count);

    return count;
}

int main() {
    int I;
    while (scanf("%d", &I) != EOF && I != 0) {
        count_parity(I);
    }

    return 0;
}
*/
#include<stdio.h>
char a[10000000];
int main()
{
    unsigned long long x,b,c;
    while(scanf("%llu",&x)==1 && x!=0)
    {
        b=0,c=0;
        printf("The parity of ");
        while(x!=0)
        {
            a[c]=x%2+'0';
            if(a[c]=='1') b++;
            x/=2;
            c++;
        }
        while(c--)printf("%c",a[c]);
        printf(" is %llu (mod 2).\n",b);
    }
    return 0;
}

File: uva10931/uva10931.py
# UVA 10931 - 奇偶性

def count_parity(I):
    binary = bin(I)[2:]  # 轉換為二進制並去掉前綴'0b'
    #bin(I): 这个内置函数将整数 I 转换为二进制字符串。例如，如果 I 是 10，bin(I) 会返回 '0b1010'。
    #[2:]: 这是一个字符串切片操作。它从字符串的第三个字符开始（索引为2）截取到末尾。这样做的目的是去掉二进制字符串前面的 '0b' 前缀。
    parity = binary.count('1')  # 計算1的數量
    return binary, parity

def main():
    while True:
        I = int(input())
        if I == 0:
            break
        #二進制, 奇偶性 = 計算奇偶性(I)
        binary,parity = count_parity(I)
        #count_parity(I): 这是对 count_parity 函数的调用，传入参数 I（用户输入的整数）。
        #binary, parity = ...: 这是Python中的多重赋值（或称为序列解包）。它假设 count_parity 函数返回两个值，并将这两个值分别赋给 binary 和 parity 变量。
        print(f"The parity of {binary} is {parity} (mod 2).")

if __name__ == "__main__":
    main()
